client/app/iplb/zone/iplz-zone-picker.component.js

```diff
@@ -0,0 +1,46 @@
+angular.module("managerApp")
+    .component("iplbZonePicker", {
+        template: `
+            <div data-ng-repeat="(country, zones) in $ctrl.groupedZones | orderHashByKey track by $index">
+                <p data-ng-bind="country"></p>
+                <oui-checkbox 
+                    on-change="$ctrl.onSelectionChanged(zone, modelValue)"
+                    data-disabled="zone.selectable.value === false"
+                    data-description="{{ zone.selectable.value === false ? zone.selectable.reason : '' }}"
+                    data-ng-repeat="zone in zones | orderBy: 'microRegion.text' track by $index" 
+                    text="{{ zone.microRegion.text }}"></oui-checkbox>
+            </div>`,
+        controller:
+            class {
+                constructor ($scope) {
+                    this._groupZones();
+                    this.selections = [];
+
+                    $scope.$watchCollection(() => this.zones, () => {
```

Comment:
$watch :( ^^
No magical alternative though... ($doCheck or copy the original object...)

Reply:
I didn't have much choice... I wanted to make a group by filter to avoid that but it just wouldn't work...